src/main/scala/com/campudus/tableaux/controller/MediaController.scala

```diff
@@ -146,8 +146,31 @@ class MediaController(override val config: TableauxConfig,
     upload.streamToFile(filePath.toString())
   }
 
-  def changeFile(uuid: UUID, title: MultiLanguageValue[String], description: MultiLanguageValue[String], externalName: MultiLanguageValue[String], folder: Option[FolderId]): Future[ExtendedFile] = {
-    fileModel.update(File(uuid, title, description, externalName, folder)).map(ExtendedFile)
+  def changeFile(uuid: UUID, title: MultiLanguageValue[String], description: MultiLanguageValue[String], externalName: MultiLanguageValue[String], internalName: MultiLanguageValue[String], mimeType: MultiLanguageValue[String], folder: Option[FolderId]): Future[ExtendedFile] = {
+    def checkInternalName(internalName: String): Future[Unit] = {
+      val p: Promise[Unit] = Promise()
+      vertx.fileSystem().exists((uploadsDirectory / Path(internalName)).toString(), {
+        case Success(java.lang.Boolean.TRUE) => p.success(())
+        case Success(java.lang.Boolean.FALSE) => p.failure(InvalidRequestException(s"File with internal name $internalName does not exist"))
+        case Failure(ex) => p.failure(ex)
+        case _ => p.failure(UnknownServerException("Error in vertx filesystem exists check"))
+      }: Try[java.lang.Boolean] => Unit)
+
+      p.future
+    }
+
+    Future.sequence(internalName.values.map { x =>
+      val internalFileName = x._2
+      if (!internalFileName.split("[/\\\\]")(0).equals(internalFileName) ||
```

Comment:
Why `[/\\\\]` and not only `[/\\]` ?
Would be good to add a test for `/something` and `\something` for example...